src/forms/components/UserForm/index.js

```diff
@@ -0,0 +1,88 @@
+import React, {useState, Fragment} from 'react';
+import {Text, TouchableOpacity} from 'react-native';
+import {Field, reduxForm} from 'redux-form';
+import {useDispatch} from 'react-redux';
+
+import {createFields} from '../../utils';
+import {USER_FORM} from '../../constants';
+
+import actionsCreator from '../../../redux/auth/actions';
+import {required, email, minChar, checkPass} from '../../validate';
+
+import {styles} from './style';
+
+const UserForm = ({navigation, ...props}) => {
+  const [changeForm, setChangeForm] = useState(true);
+  const dispatch = useDispatch();
+
+  const {handleSubmit, reset, submitting} = props;
+
+  const submitLogin = values => {
+    dispatch(actionsCreator.loginToken(navigation, values));
+  };
+
+  const submitRegister = values => {
+    dispatch(actionsCreator.registerToken(navigation, values));
+  };
+
+  const submit = changeForm ? submitLogin : submitRegister;
+  const buttonTxt = changeForm ? 'Login' : 'Register';
+
+  const handleForm = () => {
+    setChangeForm(!changeForm);
+    reset();
+  };
+
+  const submitButton = values => {
+    submit(values);
+    setTimeout(() => setChangeForm(true), 2000);
+  };
+
+  const textForm = changeForm
+    ? 'You not already an account? Create one'
+    : 'Already have an account? Login';
```

Comment:
Not getting this at all. Mind explaining your process thought about how you are handling registration/login??

Reply:
Yes, the setTimeout function is here to return the form in the initial state, showing the login input when the user logs out

Reply:
Otherwise, if a new user is registered, when they log out, they will appear again with the registration view.

Reply:
I deleted this because it was unnecessary code